src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,698 @@
+    }
+
+    /// @inheritdoc IFundingManager_v1
+    /// @dev Only payment clients can transfer orchestrator tokens
+    /// @param to_ The recipient address
+    /// @param amount_ The amount to transfer
+    function transferOrchestratorToken(address to_, uint amount_)
+        external
+        onlyPaymentClient
+    {
+        token().safeTransfer(to_, amount_);
+
+        emit TransferOrchestratorToken(to_, amount_);
+    }
+
+    /// @notice Sets fee for sell operations
+    /// @dev Only orchestrator admin can call
+    /// @param fee_ New fee amount
+    function setSellFee(uint fee_)
+        public
+        override(RedeemingBondingCurveBase_v1, IRedeemingBondingCurveBase_v1)
+        onlyOrchestratorAdmin
+    {
+        // Check that fee doesn't exceed maximum allowed
+        if (fee_ > _maxSellFee) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_FeeExceedsMaximum(
+                fee_, _maxSellFee
+            );
+        }
+
+        super._setSellFee(fee_);
+    }
+
+    /// @notice Gets current sell fee
+    /// @return sellFee_ The current sell fee
+    function getSellFee() public pure returns (uint sellFee_) {
+        return sellFee_;
+    }
+
+    /// @notice Sets fee for buy operations
+    /// @dev Only orchestrator admin can call
+    /// @param fee_ New fee amount
+    function setBuyFee(uint fee_)
+        external
+        override(BondingCurveBase_v1)
+        onlyOrchestratorAdmin
+    {
+        // Check that fee doesn't exceed maximum allowed
+        if (fee_ > _maxBuyFee) {
+            revert Module__FM_PC_ExternalPrice_Redeeming_FeeExceedsMaximum(
+                fee_, _maxBuyFee
+            );
+        }
+
+        super._setBuyFee(fee_);
+    }
+
+    /// @notice Gets current buy fee
+    /// @return buyFee_ The current buy fee
+    function getBuyFee() public pure returns (uint buyFee_) {
+        return buyFee_;
+    }
+
+    /// @notice Gets the maximum fee that can be charged for buy operations
+    /// @return maxBuyFee_ The maximum fee percentage
+    function getMaxBuyFee() public view returns (uint maxBuyFee_) {
+        return _maxBuyFee;
+    }
+
+    /// @notice Gets the maximum fee that can be charged for sell operations
+    /// @return maxSellFee_ The maximum fee percentage
+    function getMaxSellFee() public view returns (uint maxSellFee_) {
+        return _maxSellFee;
+    }
+
+    /// @notice Sets the direct operations only flag
+    /// @dev Only orchestrator admin can call
+    /// @param isDirectOperationsOnly_ The new value for the flag
+    function setIsDirectOperationsOnly(bool isDirectOperationsOnly_)
+        public
+        onlyOrchestratorAdmin
+    {
+        _isDirectOperationsOnly = isDirectOperationsOnly_;
+    }
+
+    /// @notice Gets the direct operations only flag
+    /// @return isDirectOperationsOnly_ The current value of the flag
+    function isDirectOperationsOnly() public view returns (bool) {
```

Comment:
The return value is not named in the function so far.

Reply:
modified